src/frontend/src/test-e2e/util.ts

```diff
@@ -0,0 +1,209 @@
+import { remote } from "webdriverio";
+import { command } from "webdriver";
+import * as SeleniumStandalone from "selenium-standalone";
+import { ChildProcess } from "selenium-standalone";
+
+export async function runInBrowser(
+  test: (browser: WebdriverIO.Browser) => Promise<void>
+): Promise<void> {
+  await runInBrowserCommon(true, test);
+}
+
+export async function runInNestedBrowser(
+  test: (browser: WebdriverIO.Browser) => Promise<void>
+): Promise<void> {
+  await runInBrowserCommon(false, test);
+}
+
+export async function startWebdriver(): Promise<ChildProcess | undefined> {
+  let webdriverProcess: ChildProcess | undefined;
+  let retryCount = 0;
+  let error;
+  while (!webdriverProcess && retryCount < 10) {
+    try {
+      error = undefined;
+      webdriverProcess = await SeleniumStandalone.start();
+    } catch (e) {
+      // port may still be used from previous stopped webdriver, try again
+      error = e;
+      retryCount++;
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
+  }
+  if (error !== undefined) {
+    console.warn(
+      'selenium could not be started. Make sure you installed the required webdrivers ("install-webdrivers")'
+    );
+    console.error(error);
+  }
+  return webdriverProcess;
+}
+
+export async function runInBrowserCommon(
+  outer: boolean,
+  test: (browser: WebdriverIO.Browser) => Promise<void>
+): Promise<void> {
+  let webdriverProcess;
+  if (outer) {
+    webdriverProcess = await startWebdriver();
+  }
+
+  const browser = await remote({
+    capabilities: {
+      browserName: "chrome",
+      "goog:chromeOptions": {
+        args: [
+          "--no-sandbox",
```

Comment:
why disable the sandbox?

Reply:
I copied that from somewhere but it doesn't seem necessary. Good point, I'll remove it :-)